numpy/core/src/umath/ufunc_object.c

```diff
@@ -125,6 +125,64 @@ PyUFunc_clearfperr()
     npy_clear_floatstatus();
 }
 
+/*
+ * This function analyzes the input arguments and determines an appropriate
+ * method (__array_prepare__ or __array_wrap__) function to call, taking it
+ * from the input with the highest priority.
+ */
+static PyObject*
+_find_array_method(PyObject *args, int nin, PyObject *method_name)
+{
+    int i, np;
```

Comment:
`np` as a name seems like a remnant from `prepare` - maybe rename to `n_methods`?

Reply:
Done